org.eclipse.dawnsci.nexus/src/org/eclipse/dawnsci/nexus/NexusUtils.java

```diff
@@ -799,4 +802,40 @@ public static Object extractScalarFromDataset(Object value) {
 		}
 		return null;
 	}
+
+	/**
+	 * Append data to a dataset in group node. If data node does not exist then create one. This assumes
+	 * that data is of a consistent rank R and will be appended to a shape with rank R+1 along the first
+	 * dimension 
+	 * @param f Nexus file
+	 * @param g group node
+	 * @param a dataset
+	 * @return data node
+	 * @throws NexusException
+	 * @throws DatasetException 
+	 */
+	public static DataNode appendData(NexusFile f, GroupNode g, Dataset a) throws NexusException, DatasetException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)